sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs

```diff
@@ -0,0 +1,333 @@
+
+            // Post import entities
+            var importEntitiesOp = await sapDiscoverySiteResource.ImportEntitiesAsync(WaitUntil.Completed);
+
+            Assert.IsNotNull(
+                await TrackTillConditionReachedForAsyncOperationAsync(
+                    new Func<Task<bool>>(async () => await this.TrackForDiscoveryExcelInputSasUriAsync(
+                        importEntitiesOp)), 300),
+                "SAS Uri generation failed");
+
+            var opStatus = await importEntitiesOp.UpdateStatusAsync();
+            var map = JObject.Parse(opStatus.Content.ToString());
+            var inputExcelSasUri = map?["properties"]?["discoveryExcelSasUri"].ToString();
+
+            //Upload here
+            using (var stream = File.OpenRead(@"TestData\ExcelSDKTesting.xlsx"))
+            {
+                // Construct the blob client with a sas token.
+                var blobClient = this.GetBlobContentClient(inputExcelSasUri);
+
+                await blobClient.UploadAsync(stream, overwrite: true);
+            }
+
+            Assert.IsTrue(
+                await TrackTillConditionReachedForAsyncOperationAsync(
+                    new Func<Task<bool>>(async () => await this.ValidateExcelParsingStatusAsync(
+                        43,
+                        47,
+                        importEntitiesOp)), 300),
+                "Excel Upload failed.");
+
+            var collection = sapDiscoverySiteResource.GetSAPInstances();
+
+            List<SAPInstanceData> list = new List<SAPInstanceData>();
+            // invoke the operation and iterate over the result
+            await foreach (SAPInstanceResource item in collection.GetAllAsync())
+            {
+                list.Add(item.Data);
+            }
+
+            Assert.IsNotNull(list);
+
+            var jsonString = File.ReadAllText(@"TestData\ExpectedGetSapInstanceList.json").Replace("__SapDiscoverySIteId__", sapDiscoverySiteResource.Id.ToString());
+            var json = JsonDocument.Parse(jsonString).RootElement;
+            List<SAPInstanceData> expectedList = new List<SAPInstanceData>();
+            var arr = json.EnumerateArray();
+            for (int i = 0; i < arr.Count(); i++)
+            {
+                var obj = arr.ElementAt(i);
+                expectedList.Add(SAPInstanceData.DeserializeSAPInstanceData(obj));
+            }
+
+            Assert.AreEqual(expectedList.Count(), list.Count());
+
+            list = list.OrderBy(instance => instance.Id.ToString().ToLower()).ToList();
+            expectedList = expectedList.OrderBy(instance => instance.Id.ToString().ToLower()).ToList();
+            for (int i = 0; i < list.Count() ; i++)
+            {
+                Assert.AreEqual(expectedList[i].Location, list[i].Location);
+                Assert.AreEqual(expectedList[i].LandscapeSid, list[i].LandscapeSid);
+                Assert.AreEqual(expectedList[i].Name, list[i].Name);
```

Comment:
can we use https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.comparer-1?view=net-8.0

Reply:
Done